3.2/root/usr/share/container-scripts/mongodb/init-petset-replset.sh

```diff
@@ -0,0 +1,185 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source "${CONTAINER_SCRIPTS_PATH}/common.sh"
+
+info "Waiting for local MongoDB to accept connections ..."
+wait_for_mongo_up &>/dev/null
+
+# This is a full hostname that will be added to replica set
+# (for example, "replica-2.mongodb.myproject.svc.cluster.local")
+readonly MEMBER_HOST="$(hostname -f)"
+
+# Description of possible statuses: https://docs.mongodb.com/manual/reference/replica-states/
+readonly WAIT_PRIMARY_STATUS='
+  while (rs.status().startupStatus || (rs.status().hasOwnProperty("myState") && rs.status().myState != 1)) {
+    printjson(rs.status());
+    sleep(1000);
+  };
+  printjson(rs.status());
+'
+readonly WAIT_SECONDARY_STATUS="
+  var mbrs;
+  while (!mbrs || mbrs.length == 0 || mbrs[0].state != 2) {
+    printjson(rs.status());
+    sleep(1000);
+    mbrs = rs.status().members.filter(function(el) {
+      return el.name.indexOf(\"${MEMBER_HOST}:\") > -1;
+    });
+  };
+  print(mbrs[0].stateStr);
+"
+readonly FIND_PRIMARY_HOST='
+  var name = rs.status().members.filter(function(el) {
+    return el.stateStr == "PRIMARY";
+  })[0].name;
+  print("PRIMARY|" + name);
+'
+readonly ADD_MEMBER="
+  var members = rs.status().members.filter(function(el) {
+    return el.name.indexOf(\"${MEMBER_HOST}:\") > -1;
+  });
+  if (members.length == 0) {
+    printjson(rs.add('${MEMBER_HOST}'));
+  } else {
+    print('{ \"ok\" : 1 }');
```

Comment:
When secondary joins to the replica set it executes `rs.add()` command (first branch) that return this output. When secondary was restarted and tries to join to replica set, we checking that this node already in the config and print this "fake" output manually. For the logic that is handling mongo command's output these 2 cases looks identical.